File: scala/digitalwallet/app/src/main/scala/ports/LedgerDatabase.scala
package ports

import core.domain.model.{CreateJournalEntry, LedgerQuery}

import java.time.LocalDateTime

trait LedgerDatabase {
  def insertJournalEntry(createJournalEntry: CreateJournalEntry, postedAt: Option[LocalDateTime] = None): Unit
  def getBalance(walletId: String, ledgerQueries: List[LedgerQuery]): BigDecimal
}

File: kotlin/digital-wallet/src/main/kotlin/digitalwallet/ports/LedgerDatabase.kt
package digitalwallet.ports

import digitalwallet.core.domain.models.CreateJournalEntry
import digitalwallet.core.domain.models.LedgerQuery
import java.math.BigDecimal
import java.time.LocalDateTime

interface LedgerDatabase {
    fun insertJournalEntry(createJournalEntry: CreateJournalEntry, postedAt: LocalDateTime? = null)
    fun getBalance(walletId: String, ledgerQueries: List<LedgerQuery>): BigDecimal
}